pennylane/measurements.py

```diff
@@ -536,7 +543,10 @@ def numeric_type(self):
         Returns:
             type: This is always ``int``.
         """
-        return int
+        if self.return_type is Shadow:
+            return int
+
+        return float
```

Comment:
I have to admit I'm a bit confused by what is happening here!

Reply:
`self.return_type` is either `Shadow` for the raw shadow measurement, or `ShadowExpval` for the estimation of the expectation values. In the first case the return dtype is an integer, and in the second case it is a real number.